include/nil/blueprint/input_reader.hpp

```diff
@@ -429,12 +429,20 @@ namespace nil {
                     bool is_private = current_arg->hasAttribute(llvm::Attribute::PrivateInput);
 
                     if (is_private) {
+                        if (private_input.size() == 0) {
```

Comment:
Let's move it inside the next `if` (it will also be satisfied on zero input). This way we can avoid checking the size on each loop iteration. Same for public input.